Sources/SpeziHealthKit/HealthKit Extensions/HKElectrocardiogram+RelatedData.swift

```diff
@@ -19,60 +19,52 @@ extension HKElectrocardiogram {
     
     
     /// All possible `HKCategoryType`s (`HKCategoryTypeIdentifier`s) that can be associated with an `HKElectrocardiogram`.
-    public static let correlatedSymptomTypes: [HKCategoryType] = {
-        // We disable the SwiftLint force unwrap rule here as all initializers use Apple's constants.
-        // swiftlint:disable force_unwrapping
-        [
-            HKSampleType.categoryType(forIdentifier: HKCategoryTypeIdentifier.rapidPoundingOrFlutteringHeartbeat)!,
-            HKSampleType.categoryType(forIdentifier: HKCategoryTypeIdentifier.skippedHeartbeat)!,
-            HKSampleType.categoryType(forIdentifier: HKCategoryTypeIdentifier.fatigue)!,
-            HKSampleType.categoryType(forIdentifier: HKCategoryTypeIdentifier.shortnessOfBreath)!,
-            HKSampleType.categoryType(forIdentifier: HKCategoryTypeIdentifier.chestTightnessOrPain)!,
-            HKSampleType.categoryType(forIdentifier: HKCategoryTypeIdentifier.fainting)!,
-            HKSampleType.categoryType(forIdentifier: HKCategoryTypeIdentifier.dizziness)!
-        ]
-        // swiftlint:enable force_unwrapping
-    }()
+    public static let correlatedSymptomTypes: [HKCategoryType] = [
+        HKCategoryType(.rapidPoundingOrFlutteringHeartbeat),
+        HKCategoryType(.skippedHeartbeat),
+        HKCategoryType(.fatigue),
+        HKCategoryType(.shortnessOfBreath),
+        HKCategoryType(.chestTightnessOrPain),
+        HKCategoryType(.fainting),
+        HKCategoryType(.dizziness)
+    ]
     
     
     /// Load the symptoms of an `HKElectrocardiogram` instance from an `HKHealthStore` instance.
     /// - Parameter healthStore: The `HKHealthStore` instance that should be used to load the `Symptoms`.
     /// - Returns: The symptoms associated with an `HKElectrocardiogram`.
     public func symptoms(from healthStore: HKHealthStore) async throws -> Symptoms {
-        let predicate = HKQuery.predicateForObjectsAssociated(electrocardiogram: self)
-        
-        try await healthStore.requestAuthorization(toShare: [], read: Set<HKObjectType>(HKElectrocardiogram.correlatedSymptomTypes))
-        
-        var symptoms: Symptoms = [:]
-        
-        if symptomsStatus == .present {
-            for sampleType in HKElectrocardiogram.correlatedSymptomTypes {
-                guard let sample = try await healthStore.sampleQuery(for: sampleType, withPredicate: predicate).first,
-                      let categorySample = sample as? HKCategorySample else {
-                    continue
+        switch symptomsStatus {
+        case .present:
+            // TODO ideally, the auth reques should be moved out of here!!!
+            // instead, have it run as part of the HealthKit config step?!
```

Comment:
Agree, maybe correlated identifies could be passed in the configuration options for the ECG HealthKit samples configuration object? Maybe with a default list of all symptoms and a default value that doesn't present them in addition to the ECG data type on authorization request?

Reply:
We already have a mechanism where you can `RequestReadAccess` to eg the blood pressure correlation type (which isn't actually allowed by HealthKit), and it'll automatically transform that into requests for the systolic and diastolic quantity types (which is allowed); we should probably adopt the same approach for this.

Reply:
i didn't end up implementing this; in the current PR bc it'd probably require a non-trivial amount of changes. i'll open an issue for this and fix it in a subsequent PR.